### PR TITLE
Add FIPS enabled image warning

### PR DESCRIPTION
**Pull request by mmartinv**
Show a warning message when building a FIPS enabled image in a non enabled FIPS host

### CODE REVIEW

**Comment by bcl** [comment]
I don't understand the reason for this -- if the resulting image is FIPS compliant, why print the warning? Also, long text strings like this may need to be translated, something which we don't support yet.

**Comment by 7flying** [comment]
> I don't understand the reason for this -- if the resulting image is FIPS compliant, why print the warning? Also, long text strings like this may need to be translated, something which we don't support yet.

This disclaimer is a recommendation that we got from the Crypto team so that the users know that if they're using our infra they shouldn't be generating keys (since composer doesn't run in FIPS enabled runners).

**Comment by miabbott** [comment]
> This disclaimer is a recommendation that we got from the Crypto team so that the users know that if they're using our infra they shouldn't be generating keys (since composer doesn't run in FIPS enabled runners).

I believe the context for this warning comes from this comment in the associated Jira story - https://issues.redhat.com/browse/THEEDGE-3735?focusedId=23577973&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23577973

cc:  @mrguitar @neverpanic @evgenyz

**Comment by evgenyz** [comment]
Yep. The UI/UX implementation could be different of course, as long as users understand the implications about generated cryptographic artifacts.

**Comment by neverpanic** [comment]
Could we not make this warning dynamic, i.e., depend on whether the host is actually running in FIPS mode? That would reduce the noise in situations where somebody is actually running image builder on a FIPS-enabled host.

We can check `/proc/sys/crypto/fips_enabled` for this; if it doesn't contain `1`, the host isn't in FIPS mode.

**Review by neverpanic — APPROVED** [review]
lg2m now!

**Review by achilleas-k — DISMISSED** [review]
LGTM.  I'll hold off on merging until @bcl approves since he commented.

**Comment by 7flying** [comment]
@mmartinv LGTM too, no more tests needed

File: pkg/distro/rhel9/distro_test.go
Comment (bcl) [inline pkg/distro/rhel9/distro_test.go:507]: copy pasta typo :)

Unrelated, but my brain sees these tests with almost exactly the same code and wishes it could be a function. But not really asking for that.

File: pkg/distro/fedora/imagetype.go
Comment (bcl) [inline pkg/distro/fedora/imagetype.go:397]: I think this also needs a log() like the RHEL versions. Unless there is a reason not to log on Fedora?

File: pkg/distro/rhel9/distro_test.go
Comment (mmartinv) [reply pkg/distro/rhel9/distro_test.go]: @bcl  I moved the tests to `pkg/distro/distro_test.go` and simplified some things. I hope this is ok with you.

File: pkg/distro/fedora/imagetype.go
Comment (mmartinv) [reply pkg/distro/fedora/imagetype.go:397]: There's no log in the whole function so I kept it the same way.

Comment (bcl) [reply pkg/distro/fedora/imagetype.go:397]: I looked a bit deeper and I think no logging is ok for a warning. It's really meant to make it back to the user, who isn't going to read the logs. In most cases the warnings are currently ignored, but that can be fixed later. FWIW it does look like it makes it back up to the weldr API call and is returned as part of the compose start response.

**Review by bcl — APPROVED** [review]
Looks good, thanks!

**Pull request merged**
